src/main/java/de/tum/in/www1/artemis/domain/Submission.java

```diff
@@ -138,13 +138,9 @@ public Result getFirstResult() {
      *
      * @param result
      */
-    public void replaceLatestOrIfEmptyAddResult(Result result) {
-        if (this.results.isEmpty()) {
-            this.results.add(result);
-        }
-        else {
-            results.set(results.size() - 1, result);
-        }
+    public void addResult(Result result) {
+        this.results.add(result);
+        // At the moment only one result in results is allowed
         assert results.size() < 2;
```

Comment:
I have never seen this in production java code before. I also found this online. Could you maybe throw a normal Exception here? Or is there a specific reason you used assert here that I miss.
https://en.wikibooks.org/wiki/Java_Programming/Keywords/assert#:~:text=assert%20is%20a%20Java%20keyword,runtime%20system%20throws%20an%20AssertionError%20.
`Assertions are usually used as a debugging aid. They should not be used instead of validating arguments to public methods, or in place of a more precise runtime error exception.`

Reply:
We changed it to throw an Exception. This check will be removed later, as with second corrections enabled we will need longer Submission.results lists.